pinecone/index_connection.go

```diff
@@ -144,11 +335,62 @@ type QueryByVectorValuesRequest struct {
 	SparseValues    *SparseValues
 }
 
+// QueryVectorsResponse holds the parameters for the QueryVectorsResponse object,
+// which is returned by the QueryByVectorValues method.
+//
+// Fields:
+//  - Matches: The vectors (nearest neighbors) that are most similar to the query vector.
+//  - Usage: The usage information for the request.
 type QueryVectorsResponse struct {
 	Matches []*ScoredVector `json:"matches,omitempty"`
 	Usage   *Usage          `json:"usage,omitempty"`
 }
 
+// QueryByVectorValues queries a Pinecone index for vectors (
+// nearest neighbors) that are most similar to a provided query vector.
+//
+// Returns a pointer to a QueryVectorsResponse object and an error if the request fails.
+//
+// Parameters:
+//  - ctx: A context.Context object controls the request's lifetime,
+//  - allowing for the request to be canceled or to timeout according to the context's deadline.
+//   in: A QueryByVectorValuesRequest object with the parameters for the request.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//	  ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//	  SourceTag: "your_source_identifier", // optional
+//	}
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//	if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//  }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//
+//  idx, err := pc.Index(idxs[0].Host)
+//  if err != nil {
+//	  fmt.Println("Error:", err)
+//  }
+//
+//  queryVector := []float32{1.0, 2.0}
+//	topK := uint32(10)
+//
+//	res, err := idx.QueryByVectorValues(ctx, &pinecone.QueryByVectorValuesRequest{
+//	  Vector:        queryVector,
+//	  TopK:          topK,  // number of vectors (nearest neighbors) you want returned
```

Comment:
Should we be referencing "nearest neighbors"? I don't think we do in other SDKs, and it may be superfluous here. Isn't it based on the similarity metric anyways?

It's in a few places in the comments in this file, I'm thinking we should maybe just drop it.

Reply:
Yeah can do!